src/ng-jest-transformer.ts

```diff
@@ -79,9 +79,11 @@ export class NgJestTransformer extends TsJestTransformer {
             this.#ngJestLogger.debug({ filePath }, 'process with esbuild');
 
             const compilerOpts = configSet.parsedTsConfig.options;
+            const useESM = Boolean(transformOptions.supportsStaticESM && configSet.useESM);
```

Comment:
```suggestion
            const useESM = transformOptions.supportsStaticESM && configSet.useESM;
```
should be good enough, we don't need to explicit casting to `Boolean` :)

Reply:
I didn't put it at first either, but it turns out in falsy case it's not `false`, but `undefined`. This will have to be added to the mocks, and in my opinion, it will look ugly. Plus, I'm not sure that esbuild won't eventually react to `undefined` differently. But if it's necessary, tell me, I'll remove it.

Reply:
How about we do
```
const useESM = transformOptions.supportsStaticESM && configSet.useESM;

const { code, map } = transformSync(fileContent, {
                loader: 'js',
                format: transformOptions.supportsStaticESM && configSet.useESM ? 'esm' : 'cjs',
                format: useESM ? 'esm' : 'cjs',
                supported: useESM ? { 'dynamic-import': true } : undefined,
```

Reply:
Ah, no, that's not good, we need to explicitly pass `false`. I'll do the opposite.

```js
supported: useESM ? undefined : { 'dynamic-import': false },`
```